src/handlers/channel_item_handler.erl

```diff
@@ -99,6 +100,8 @@ from_json(Req, State = #{is_new := false,
   end.
 
 delete_resource(Req, State = #{email := Email, name := Name}) ->
+  {ok, Channel} = db_channel:get(Email, Name),
```

Comment:
this isn't necessary, the channel is already part of the State (check the rest of the code).